includes/media.php

```diff
@@ -410,26 +410,30 @@ function pods_image_resize( $attachment_id, $size ) {
  *
  * @since 2.5.0
  *
- * @param string|array|int $url  Can be a URL of the source file, or a Pods audio field.
+ * @param string|array|int $url  The URL string, an array of post information, or an attachment ID.
  * @param bool|array       $args Optional. Additional arguments to pass to wp_audio_shortcode().
  *
  * @return string
  */
 function pods_audio( $url, $args = false ) {
 
-	if ( ! is_string( $url ) ) {
-		$id = $url;
-		if ( ! is_numeric( $id ) ) {
-			$id = pods_v( 'ID', $url );
-		}
-		$url = wp_get_attachment_url( $id );
+	// Support arrays.
+	if ( is_array( $url ) ) {
+		$url = pods_v( 'ID', $url );
+	}
+
+	// Support IDs.
+	if ( false !== strpos( $url, '://' ) && is_numeric( $url ) ) {
```

Comment:
@sc0ttkclark  Why not just check for `is_numeric`? It's faster and makes the strpos redundant.

Reply:
that was supposed to be a ||, will fix

```suggestion
	if ( is_numeric( $url ) || false !== strpos( $url, '://' ) ) {
```